library/Fission/Web/User/Create.hs

```diff
@@ -28,29 +30,46 @@ type PasswordAPI
   :> PostCreated '[JSON] ()
 
 withDID ::
-  ( MonadDNSLink m
-  , MonadLogger  m
-  , MonadTime    m
-  , User.Creator m
+  ( MonadDNSLink      m
+  , MonadLogger       m
+  , MonadTime         m
+  , MonadEmail        m
+  , User.Creator      m
+  , Challenge.Creator m
   )
   => DID
   -> ServerT API m
 withDID DID {..} User.Registration {username, email} = do
   now <- currentTime
-  Web.Err.ensureM $ User.create username publicKey email now
-  return NoContent
+  userId <- Web.Err.ensureM $ User.create username publicKey email now
+  challenge <- Web.Err.ensureM $ Challenge.create userId
+
+  sendVerifyEmail (Recipient email username) challenge >>= \case
+    Left _ ->
+      Web.Err.throw err500 { errBody = "Could not send verification email" }
+
+    Right _ -> 
+      return NoContent
 
 withPassword ::
-  ( MonadDNSLink m
-  , MonadLogger  m
-  , MonadTime    m
-  , User.Creator m
+  ( MonadDNSLink      m
+  , MonadLogger       m
+  , MonadTime         m
+  , MonadEmail        m
+  , User.Creator      m
+  , Challenge.Creator m
   )
   => ServerT PasswordAPI m
 withPassword User.Registration {password = Nothing} =
   Web.Err.throw err422 { errBody = "Missing password" }
 
 withPassword User.Registration {username, password = Just pass, email} = do
   now <- currentTime
-  Web.Err.ensureM $ User.createWithPassword username pass email now
-  return ()
+  userId <- Web.Err.ensureM $ User.createWithPassword username pass email now
+  challenge <- Web.Err.ensureM $ Challenge.create userId
+
+  sendVerifyEmail (Recipient email username) challenge >>= \case
+    Left _ -> 
+      Web.Err.throw err500 { errBody = "Could not send verification email" }
+    Right _ -> 
```

Comment:
Oh crap, right. Password registration is for Heroku only. We need to figure this out how to get a user email from them. Otherwise this will break the Jeroku add on!

Reply:
oh yeahhhh good call. we could just not verify email? and auto-verify the user if it was created through heroku

Reply:
On that note, should we even keep around this password registration route? Heroku doesn't use it, it uses `POST /heroku/resource`

Reply:
Yeah, I realized that much later 😛 

I guess the actual question is if we’re able to require an email in the database until we’ve figured the Heroku stuff out.

The password-based endpoint is deprecated. Let’s leave it there for now. I’d rather not cut it until we have a major version bump.